apps/meteor/client/components/message/variants/RoomMessage.tsx

```diff
@@ -69,8 +69,9 @@ const RoomMessage = ({ message, sequential, all, mention, unread, context, ignor
 		>
 			<MessageLeftContainer>
 				{!sequential && message.u.username && !selecting && (
-					<UserAvatar
-						url={message.avatar}
+					<MessageAvatar
+						emoji={message.emoji}
```

Comment:
why not keeping this explicit?

```
{emoji && <EmojiAvatar />}
{!emoji && <UserAvatar />}
```
?

Reply:
Because I would have to replicate this same check in many places (threads, threadlist, discussion list, thread preview, etc). Doing this I keep this logic contained in a single place, which IMO makes it easier to maintain in case we change this.

I'm happy to change this if you think it would be better.